lib/python/qmk/commands.py

```diff
@@ -61,3 +64,21 @@ def parse_configurator_json(configurator_file):
     user_keymap = json.load(configurator_file)
 
     return user_keymap
+
+
+def run(command, *args, **kwargs):
+    """Run a command with subprocess.run
+    """
+    OS = platform.platform().lower()  # noqa (N806), uppercase name is ok in this instance
+
+    if isinstance(command, str):
+        raise TypeError('command must be a sequence of items such as list or tuple.')
+
+    if 'windows' in OS:
+        command = map(shlex.quote, command)
```

Comment:
Two things:

1. Something like `platform_id` would be preferable to `OS`: it's more descriptive, it better matches what the function returns, and it doesn't violate PEP8.
2. The error raised on line 75 doesn't exactly match what's being checked. In this case, I think it's better to abide by the “ask for forgiveness, not permission” principle; i.e. see if it quacks like a duck, and fail gracefully if it doesn't.

```suggestion
    platform_id = platform.platform().lower()
    if 'windows' in platform_id:
        try:
            command = map(shlex.quote, command)
        except TypeError as e:
            raise TypeError("command must be an iterable such as list or tuple") from e
```

Since calling `map` on a non-iterable type raises a `TypeError` in and of itself, if the error message isn't important the `raise` isn't necessary and this boils down to:

```suggestion
    platform_id = platform.platform().lower()
    if 'windows' in platform_id:
        command = map(shlex.quote, command)
```

In case the explicit check is preferred over EAFP for reasons of clarity, it should be moved to the top of the block and its condition changed to better reflect the intent of the error message (or vice-versa):

```suggestion
    if hasattr(command, '__iter__'):
        raise TypeError("command must be an iterable such as list or tuple.")

    platform_id = platform.platform().lower()
    if 'windows' in platform_id:
        command = map(shlex.quote, command)
```

Though I believe this to be the least elegant solution.

Reply:
> Since calling map on a non-iterable type raises a TypeError in and of itself, if the error message isn't important the raise isn't necessary and this boils down to:

The raise is necessary so that we have the same behavior on both windows and non-windows systems. Our intent is to disallow strings as command for all OS's, not only windows.

Using `hasattr` as you have is not equivalent because strings have an `__iter__` attribute.

Reply:
Ah, I misunderstood the intent; the check is there to avoid mapping over the string. The error message threw me off. In that case, perhaps something like this is more in order?

```suggestion
    if isinstance(command, str):
        raise TypeError("command should be a list or tuple, not a string.")

    platform_id = platform.platform().lower()
    if 'windows' in OS:
        command = map(shlex.quote, command)
```

Reply:
Thanks for the suggestions.  
You made us rethink the wording, so that's changed a little bit.